version: "3"

services:
  3x-ui:
    image: 'ghcr.io/mhsanaei/3x-ui:latest'
    container_name: 3x-ui
    hostname: vpn.automateme.ru
    volumes:
      - '/home/user/3x-ui/db:/etc/x-ui/'
      - '/home/user/3x-ui/cert:/root/cert/'
    environment:
      XRAY_VMESS_AEAD_FORCED: 'false'
    tty: true
    networks:
      - my_network
    restart: unless-stopped
    ports:
      - "4443:443"
      - "2053:2053"
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.3x-ui-http.rule=Host(`vpn.automateme.ru`) && PathPrefix(`/`) && EntryPoint(`http`)"
      - "traefik.http.routers.3x-ui-http.middlewares=redirect-to-https"
      - "traefik.http.middlewares.redirect-to-https.redirectscheme.scheme=https"
      - "traefik.http.routers.3x-ui-https.rule=Host(`vpn.automateme.ru`) && PathPrefix(`/`) && EntryPoint(`https`)"
      - "traefik.http.routers.3x-ui-https.tls.certresolver=letsencrypt"
      - "traefik.http.services.3x-ui.loadbalancer.server.port=2053"

networks:
  my_network:
    external: true
